### PR TITLE
[ENH] remove empty

### PR DESCRIPTION
**Pull request by samukweku**
Fix `remove_empty` where the dataframe is not mutated.

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1213--pyjanitor.netlify.app

File: janitor/functions/remove_empty.py [hunk 1]
```diff
@@ -37,12 +35,12 @@ def remove_empty(df: pd.DataFrame) -> pd.DataFrame:
...
-
-    nancols = df.columns[df.isna().all(axis=0)]
-    df = df.drop(columns=nancols)
-
-    return df
+    outcome = df.isna()
+    outcome = df.loc[~outcome.all(axis=1), ~outcome.all(axis=0)]
+    if reset_index:
+        return outcome.reset_index(drop=True)
+    return outcome
```
Comment (Zeroto521) [inline janitor/functions/remove_empty.py:46]: It seems need to add a new test case to cover this line.

**Review by Zeroto521 — APPROVED** [review]

File: tests/functions/test_remove_empty.py [hunk 2]
```diff
@@ -21,11 +21,24 @@ def test_remove_empty(df):
 
 @pytest.mark.functions
 def test_index_after_remove_empty():
-    # This test ensures that the indexed is reset correctly.
+    """This test ensures that the indexed is reset correctly"""
```
Comment (thatlittleboy) [inline tests/functions/test_remove_empty.py]: Just requesting a small typo fix in the test docstring.
```suggestion
    """This test ensures that the index is reset correctly"""
```

**Review by thatlittleboy — APPROVED** [review]
domino-effect-self-high-five

**Comment by thatlittleboy** [comment]
@samukweku sorry, one last thing, could you please update the CHANGELOG? I'll merge in once done. Thanks!

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
Merging! Thank you, @samukweku, for the PR, and @thatlittleboy and @Zeroto521 for reviewing!

**Pull request merged**